pallets/compliance-manager/src/lib.rs

```diff
@@ -824,38 +835,33 @@ impl<T: Trait> ComplianceManagerTrait<T::Balance> for Module<T> {
         _value: T::Balance,
         treasury_did: Option<IdentityId>,
     ) -> StdResult<u8, &'static str> {
-        let (treasury_sender, treasury_receiver) = if treasury_did.is_some() {
-            (treasury_did == from_did_opt, treasury_did == to_did_opt)
-        } else {
-            (false, false)
-        };
-        // Transfer is valid if ALL receiver AND sender rules of ANY asset rule are valid. An
-        // exception is made for the cases when either the sender or the receiver is the treasury
-        // DID, which covers issuance and redemption transactions.
+        // Transfer is valid if ALL receiver AND sender rules of ANY asset rule are valid.
         let asset_rules = Self::asset_rules(ticker);
-        if asset_rules.is_paused
-            || (asset_rules.rules.len() == 0
-                && ((from_did_opt == None && treasury_receiver)
-                    || (treasury_sender && to_did_opt == None)))
-        {
+        if asset_rules.is_paused {
             return Ok(ERC1400_TRANSFER_SUCCESS);
         }
         for active_rule in asset_rules.rules {
-            let mut rule_satisfied = false;
             if let Some(from_did) = from_did_opt {
-                rule_satisfied = treasury_sender
-                    || Self::are_all_rules_satisfied(ticker, from_did, &active_rule.sender_rules);
-                if !rule_satisfied {
+                if !Self::are_all_rules_satisfied(
```

Comment:
Does that mean the treasury_did should also possess all the valid claims as to any sender?  &  issuance will only work if the receiver has the `TargetIdentity::specific(identity)`  claim.

Reply:
`TargetIdentity::specific(identity)` is not a claim, it's a rule. The receiver needs no claims. However, there should be separate compliance manager rule that says something along the lines of "Bypass all receiver restrictions if the sender is treasury".

Reply:
I think it should be something like "Bypass all the restrictions when the sender is treasury or receiver is treasury". We still need to impose the restrictions on the receiver if the sender is the treasury or vice-versa. 

Even my main concern with the current code is that Treasury did have to pass all the rules applied to the sender itself and that defeat the purpose of giving the special case for the treasury.

Reply:
> I think it should be something like "Bypass all the restrictions when the sender is treasury or receiver is treasury". We still need to impose the restrictions on the receiver if the sender is the treasury or vice-versa.

We don't need to impose anything. The issuer can create rules as per their regulatory requirements. FWIW, I meant the rule to be "Sender is treasury". This implies "Bypass all receiver restrictions if the sender is treasury". To cover the redemption case, the issuer can add another rule saying "Receiver is treasury" (Implicit - no rules for sender).

> Even my main concern with the current code is that Treasury did have to pass all the rules applied to the sender itself and that defeat the purpose of giving the special case for the treasury.

This change removes the special case of the treasury. The issuer must add proper compliance rules before issuing. The dApp can batch both calls together for the issuer.

Reply:
I still don't understand why we need the special rule type? I can still issue asset to the treasury without having the `TargetIdentity:Treasury` rule in place.

Reply:
This rule is not to cover issuance. It is to facilitate distribution to investors after the tokens are already issued to the treasury.

This PR separates the process of issuance from distribution. Issuance is always done to the treasury and then those tokens are distributed to the investors.